source/core/TypeAndDataManager.cpp

```diff
@@ -704,13 +708,17 @@ NIError TypeCommon::CopyData(const void* pSource, void* pDest, IntIndex count)
                 break;
         }
     }
+    if (_isDataItem)
+        SetValueRefNeedsUpdate(this, pDest);
     return err;
 }
 //------------------------------------------------------------
 NIError TypeCommon::MultiCopyData(const void* pSource, void* pDest, IntIndex count)
 {
     if (IsFlat() && TopAQSize() == 1) {
         memset(pDest, (int)*(AQBlock1*)pSource, count);
+        if (_isDataItem)
```

Comment:
Should we do this in the IsFlat() case? Let's discuss.

Reply:
I'm actually removing these checks, because they are not needed and only work if we are called through the destination type.
The actual code path which implements the needed check is in the custom emit proc for the Copy instruction.